cli/pcluster/config/mapping.py

```diff
@@ -0,0 +1,530 @@
+            }),
+            ("export_path", {}),  # TODO add regex
+            ("import_path", {}),  # TODO add regex
+            ("weekly_maintenance_start_time", {}),  # TODO add regex
+        ]
+    )
+}
+
+CLUSTER = {
+    "type": ClusterSection,
+    "key": "cluster",
+    "label": "default",
+    "validator": cluster_validator,
+    "params": {
+        # Basic configuration
+        "key_name": {
+            "cfn": "KeyName",
+            "validator": ec2_key_pair_validator,
+        },
+        "template_url": {
+            # TODO add regex
+            "validator": url_validator,
+        },
+        "base_os": {
+            "default": "alinux",
+            "cfn": "BaseOS",
+            "allowed_values": ["alinux", "ubuntu1404", "ubuntu1604", "centos6", "centos7"],
+        },
+        "scheduler": {
+            "default": "sge",
+            "cfn": "Scheduler",
+            "allowed_values": ["awsbatch", "sge", "slurm", "torque"],
+            "validator": scheduler_validator,
+        },
+        "shared_dir": {
+            "type": SharedDirParam,
+            "cfn": "SharedDir",
+            "default": "/shared",
+        },
+        # Cluster configuration
+        "placement_group": {
+            "cfn": "PlacementGroup",
+            "validator": ec2_placement_group_validator,
+        },
+        "placement": {
+            "default": "compute",
+            "cfn": "Placement",
+            "allowed_values": ["cluster", "compute"],
+        },
+        # Master
+        "master_instance_type": {
+            "default": "t2.micro",  # TODO add regex or validator
+            "cfn": "MasterInstanceType",
+        },
+        "master_root_volume_size": {
+            "type": IntParam,
+            "default": 20,
+            "allowed_values": r"^([0-9]+[0-9]{2}|[2-9][0-9])$",  # >= 20
+            "cfn": "MasterRootVolumeSize",
+        },
+        # Compute fleet
+        "compute_instance_type": {
+            "default": "t2.micro",
+            "cfn": "ComputeInstanceType",
+            "validator": compute_instance_type_validator,
+        },
+        "compute_root_volume_size": {
+            "type": IntParam,
+            "default": 20,
+            "allowed_values": r"^([0-9]+[0-9]{2}|[2-9][0-9])$",  # >= 20
+            "cfn": "ComputeRootVolumeSize",
+        },
+        "initial_queue_size": {
+            "type": DesiredSizeParam,
+            "default": 0,
+            "cfn": "DesiredSize",  # TODO verify the update case
+        },
+        "max_queue_size": {
+            "type": MaxSizeParam,
+            "default": 10,
+            "cfn": "MaxSize",
+            "validator": None,  # TODO we could test the account capacity
+        },
+        "maintain_initial_size": {
+            "type": MaintainInitialSizeParam,
+            "default": False,
+            "cfn": "MinSize",
+        },
+        "min_vcpus": {
+            "type": MinSizeParam,
```

Comment:
Can Min/Max/DesiredSizeParam be grouped in a generic QueueSizeParam?

Reply:
Done. Great suggestion. 👍